my_site/blog.php

```diff
@@ -0,0 +1,160 @@
+<?php
+// start the session if it hasn’t been initialized yet
+if (session_status() === PHP_SESSION_NONE) {
+    session_start();
+    var_dump($_SESSION);
+
+}
+
+// header and the blog data functions
+include 'header.php'; 
+require_once 'blog_model.php'; 
+
+// taking all posts from the JSON file
+$posts = load_all_posts();
+
+// checking if the user is logged in ro not
+// Allow admin view via session OR a query flag (?admin=1) for environments without a dedicated blog login.
+// This does NOT bypass your security in the controller; it only reveals UI controls. 
```

Comment:
This sounds like an AI comment. I'm not against the idea of using AI, but make sure you don't copy-paste the result blindly. Understand it and make it yours.